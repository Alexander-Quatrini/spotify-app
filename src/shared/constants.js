module.exports = {
    channels: {
        CALL_API: "callAPI",
        RECEIVE_API: "receiveAPI",
        RECEIVE_USER: "receiveUser",
        RECEIVE_QUEUE: "receiveQueue",
        RECEIVE_LIBRARY: "receiveLibrary",
        SAVE_DATA: "saveData",
        SAVE_DATA_STATS: "saveDataSTATS",
        RECEIVE_FEATURES: "recieveFeatures",
        RECEIVE_CURRENTLY_PLAYING: "receiveCurrentlyPlaying",
        RATE_LIMIT_EXCEEDED: "rateLimitExceeded",
        UNKNOWN_ERROR: "unknownError",
    }
}