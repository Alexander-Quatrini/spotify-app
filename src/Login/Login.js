import React, { useEffect } from 'react';
import { channels } from '../shared/constants';

function Login(props) {

    let libraryData = null;

    useEffect(() => {
        window.api.receive(channels.RECEIVE_API, (response) =>{
            console.log(response);
            libraryData = response;
        })
    });

    return(
        <div className="login">
            <a href='http://localhost:5000/api/login' target='_blank'>Login</a>
            <input type="button" data-testid="library-load" value="Load Library!" onClick={loadLibrary}/>
        </div>
    );
}

function loadLibrary(){
    window.api.send(channels.CALL_API, {url: "/api/getlibrary", method: "GET", isBody: false, channel: channels.RECEIVE_API});
}

export default Login;