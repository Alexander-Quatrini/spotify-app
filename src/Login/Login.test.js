import Login from "./Login";
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react'
import React from "react";
import { channels } from "../shared/constants";

const mElectron = { on: jest.fn(), send: jest.fn(), receive: jest.fn()};

describe('Login', () => {

    beforeEach(() =>{
        global.window.api = mElectron;
    })
    
    it('works', () =>{
        expect.assertions(1);
        const component = renderer.create(<Login></Login>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });


    it('receives data',  () => {
        
        global.window.api.receive.mockImplementationOnce((channel, callback) => {
            callback('{}');
        })

        const spy = jest.spyOn(console, 'log')

        expect.assertions(2);
        const component = render(<Login/>)
        const button =  component.getByTestId('library-load');

        fireEvent.click(button);

        expect(global.window.api.send).toBeCalledWith(channels.CALL_API, expect.any(Object));
        expect(spy).toBeCalledWith('{}');
    });
})