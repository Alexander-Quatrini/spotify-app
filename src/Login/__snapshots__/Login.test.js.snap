// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login works 1`] = `
<div
  className="login"
>
  <a
    href="http://localhost:5000/api/login"
    target="_blank"
  >
    Login
  </a>
  <input
    data-testid="library-load"
    onClick={[Function]}
    type="button"
    value="Load Library!"
  />
</div>
`;
